tests/integration_tests/functional/test_api.py

```diff
@@ -940,52 +944,81 @@ def test_api_version(test_microvm_with_api):
     assert out.strip()[1:] == preboot_response.json()['firecracker_version']
 
 
-def test_api_vsock(test_microvm_with_api):
+def test_api_vsock(bin_cloner_path):
     """
     Test vsock related API commands.
 
     @type: functional
     """
-    test_microvm = test_microvm_with_api
-    test_microvm.spawn()
-    test_microvm.basic_config()
+    builder = MicrovmBuilder(bin_cloner_path)
+    artifacts = ArtifactCollection(_test_images_s3_bucket())
 
-    response = test_microvm.vsock.put(
-        vsock_id='vsock1',
+    # Test with the current build.
+    vm_instance = builder.build_vm_nano()
+    _test_vsock(vm_instance.vm)
+
+    # Fetch 1.0.0 and older firecracker binaries.
+    # Create a vsock device with each FC binary
+    # artifact.
+    firecracker_artifacts = artifacts.firecrackers(
+        # v1.0.0 deprecated `vsock_id`.
+        min_version="1.0.0")
+
+    for firecracker in firecracker_artifacts:
+        firecracker.download()
+        jailer = firecracker.jailer()
+        jailer.download()
+
+        vm_instance = builder.build_vm_nano(
+                                        fc_binary=firecracker.local_path(),
+                                        jailer_binary=jailer.local_path())
+
+        _test_vsock(vm_instance.vm)
+
+
+def _test_vsock(vm):
+    # Create a vsock device.
+    response = vm.vsock.put(
         guest_cid=15,
         uds_path='vsock.sock'
     )
-    assert test_microvm.api_session.is_status_no_content(response.status_code)
+    assert vm.api_session.is_status_no_content(response.status_code)
```

Comment:
I don't think we should because we might have something else deprecated in a future minor version, and when we test that binary this check will fail. As far as the current version is concerned, we just want this format of the request to work. In this functional test, we also check for the deprecated header field because we know for sure that it has to be there if we use `vsock_id`.